atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/table/common/TableTasks.java

```diff
@@ -480,4 +480,40 @@ PartialDiffStats call(RangeRequest request, MutableRange range, DiffStrategy str
         private long rowsCopied = 0;
         private long cellsCopied = 0;
     }
+
+    private static class InterruptibleRangeExecutor {
+        private ExecutorService exec;
+        private int batchSize;
+        private int threadCount;
+
+        InterruptibleRangeExecutor(ExecutorService exec, int batchSize, int threadCount) {
+            this.exec = exec;
+            this.batchSize = batchSize;
+            this.threadCount = threadCount;
+        }
+
+        void executeTask(String taskName, InterruptibleRangeTask task) throws InterruptedException {
+            BlockingWorkerPool pool = new BlockingWorkerPool(exec, threadCount);
+            for (final MutableRange range : getRanges(threadCount, batchSize)) {
+                if (Thread.currentThread().isInterrupted()) {
+                    log.info("Thread interrupted. Cancelling {} of range {}", taskName, range);
+                    break;
+                }
+                pool.submitTask(() -> {
+                    do {
+                        if (Thread.currentThread().isInterrupted()) {
+                            log.info("Thread interrupted. Cancelling {} of range {}", taskName, range);
+                            break;
+                        }
+                        try {
+                            task.execute(range);
+                        } catch (InterruptedException e) {
+                            throw Throwables.rewrapAndThrowUncheckedException(e);
```

Comment:
Generally speaking this seems suspicious (I realise it was copied from the original code) - throwing a non-`InterruptedException` on interruption seems awkward.

It seems like if the task fails, then waitForSubmittedTasks() looks like it will throw a RuntimeException of an ExecutionException of a RuntimeException of an InterruptedException. _This is the same as the past behaviour_, so I wouldn't block the PR on this, but seems pretty weird to me